### ISSUE TITLE
No Internet Freeze

### ISSUE THREAD

**Issue by card100** [GRA0007/Blue#22]
I noticed that whenever I attempted to start a block that required internet the program would freeze and I would have to reload the site again.  However, I tried the <internet connection?> block and it returned false (which I'm not sure why but that's a whole different issue).  The blocks that require an internet connection (like the WebSockets and URLs) should first question a connection to the internet and abort if there is none.

^Hopefully that made sense

**Comment by card100** [comment]
My browser / operating system: MacOS Macintosh X 10.12.6, Chrome 60.0.3112.113, Flash 27.0 (release 0)

**Comment by NoMod-Programming** [comment]
That definitely makes sense. I'm away from my programming computer, but judging by the behaviour, it is crashing. The question is what it should report if there is no internet. A blank string? And error message? That would need to be decided

**Comment by GRA0007** [comment]
Yes they definitely should. From what I can remember, the <internet connection?> block doesn't even work anyway, but I'm going to try again with this project after my final exams in two weeks. 😪
